src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -24,73 +27,34 @@
     private final Properties kafkaProperties;
     private final KafkaSettings kafkaSettings;
+    private final ScheduledExecutorService scheduledExecutor;
 
     public KafkaLocationManager(final ZooKeeperHolder zkFactory, final KafkaSettings kafkaSettings) {
         this.zkFactory = zkFactory;
-        this.kafkaProperties = buildKafkaProperties(fetchBrokers());
+        this.kafkaProperties = new Properties();
         this.kafkaSettings = kafkaSettings;
+        this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
+        this.scheduledExecutor.scheduleAtFixedRate(this::updateBootstrapServers, 0, 1, TimeUnit.MINUTES);
     }
 
-    static class Broker {
-        final String host;
-        final Integer port;
-
-        private Broker(final String host, final Integer port) {
-            this.host = host;
-            this.port = port;
-        }
-
-        static Broker fromByteJson(final byte[] data) throws JSONException, UnsupportedEncodingException {
-            final JSONObject json = new JSONObject(new String(data, "UTF-8"));
-            final String host = json.getString("host");
-            final Integer port = json.getInt("port");
-            return new Broker(host, port);
-        }
-
-        public String toString() {
-            return this.host + ":" + this.port;
-        }
-    }
-
-    private List<Broker> fetchBrokers() {
+    private void updateBootstrapServers() {
         final List<Broker> brokers = new ArrayList<>();
         try {
             final CuratorFramework curator = zkFactory.get();
             for (final String brokerId : curator.getChildren().forPath(BROKERS_IDS_PATH)) {
-                try {
-                    final byte[] brokerData = curator.getData().forPath(BROKERS_IDS_PATH + "/" + brokerId);
-                    brokers.add(Broker.fromByteJson(brokerData));
-                } catch (final Exception e) {
-                    LOG.info(String.format("Failed to fetch connection string for broker %s", brokerId), e);
-                }
+                final byte[] brokerData = curator.getData().forPath(BROKERS_IDS_PATH + "/" + brokerId);
+                brokers.add(Broker.fromByteJson(brokerData));
             }
         } catch (final Exception e) {
             LOG.error("Failed to fetch list of brokers from ZooKeeper", e);
+            return;
         }
 
-        return brokers;
-    }
-
-    private static String buildBootstrapServers(final List<Broker> brokers) {
-        final StringBuilder builder = new StringBuilder();
-        brokers.stream().forEach(broker -> builder.append(broker).append(","));
-        return builder.deleteCharAt(builder.length() - 1).toString();
-    }
-
-    private Properties buildKafkaProperties(final List<Broker> brokers) {
-        final Properties props = new Properties();
-        props.put("bootstrap.servers", buildBootstrapServers(brokers));
-        return props;
-    }
-
-    @Scheduled(fixedDelay = 30000)
```

Comment:
I remember it from a conversation but just wanted to make sure. This code was not working as expected and the brokers IPs were not refreshing. Right?